src/textarea/Textarea.tsx

```diff
@@ -0,0 +1,250 @@
+import React, { PureComponent, RefObject } from 'react'
+import cc from 'classcat'
+
+import prefix from '_utils'
+import Button, { ButtonStatus } from 'button'
+import isEmpty from 'lodash.isempty'
+
+export interface CommonFormFields {
+  name: string
+  id?: string
+  placeholder?: string
+  maxLength?: number
+  autoCorrect?: 'on' | 'off'
+  autoComplete?: 'on' | 'off'
+  disabled?: boolean
+  readOnly?: boolean
+  autoFocus?: boolean
+  required?: boolean
+  title?: string
+  onFocus?: (
+    event: React.FocusEvent<HTMLTextAreaElement>,
+  ) => void
+  onBlur?: (
+    event: React.FocusEvent<HTMLTextAreaElement>,
+  ) => void
+}
+
+type errorField = string | JSX.Element
+
+export interface TextAreaProps extends CommonFormFields {
+  defaultValue?: string
+  labelledBy?: string
+  onChange?: (obj: OnChangeParameters) => void
+  className?: Classcat.Class
+  error?: errorField
+  label?: string
+  buttonTitle?: string
+  focus?: boolean
+  pattern?: string
+  fieldRef?: (textarea: HTMLTextAreaElement) => void
+  focusBorder?: boolean
+  onButtonClick?: (event: React.MouseEvent<HTMLElement>) => void
+  buttonIcon?: JSX.Element
+  fitContent?: boolean
+}
+
+interface TextAreaAttributes extends CommonFormFields {
+  className?: string
+  value: string
+  ['aria-invalid']?: 'true' | 'false'
+  ['aria-labelledby']?: string
+  ref?: (textarea: HTMLTextAreaElement) => void
+  onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void
+}
+
+export interface TextAreaState {
+  readonly value: string
+  readonly hasFocus: boolean
+}
+
+const DisplayError = (error: errorField) => {
+  const className = 'kirk-error-message'
+
+  return React.isValidElement(error) ? (
+    React.cloneElement(error, { className } as Object)
+  ) : (
+    <span role="alert" className={className}>
+      {error}
+    </span>
+  )
+}
+
+export default class Textarea extends PureComponent<TextAreaProps, TextAreaState> {
+  private textarea: HTMLInputElement | HTMLTextAreaElement
+  static defaultProps: Partial<TextAreaProps> = {
+    fieldRef() {},
+    onFocus() {},
+    onBlur() {},
+    focusBorder: true,
+  }
+
+  state = {
+    value: this.props.defaultValue,
+    hasFocus: false,
+  }
+
+  buttonRef: HTMLButtonElement = null
+  textareaRef: RefObject<HTMLTextAreaElement> = React.createRef()
+
+  componentDidMount() {
+    if (this.textarea && this.props.focus) {
+      this.textarea.focus()
+    }
+  }
+
+  componentWillReceiveProps({ defaultValue, focus }: TextAreaProps) {
```

Comment:
Can you add a todo here, to refactor this method ? `componentWillReceiveProps` is now deprecated.

See: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Also this method is probably buggy: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-erasing-state-when-props-change

Reply:
Done. BBC-498 created too.